DesignSafe-CI/portal
# task/DES-2755 + DES-2756: Allocations backend + hooks

## Overview: ##
Adds allocations backend + client hooks for v3 apps redesign
## PR Status: ##


## Related Jira tickets: ##

* [DES-2755](https://tacc-main.atlassian.net/browse/DES-2755)
* [DES-2756](https://tacc-main.atlassian.net/browse/DES-2756)

## Summary of Changes: ##
- Added allocations API view + TAS helper function for fetching allocations by username
- Added allocations model for elasticsearch + added model to new index in `elasticsearch_settings`
- Added TASAllocation Type for client hooks
- Added `useGetAllocations` client hook for use in apps form

## Testing Steps: ##
1. Go to https://designsafe.dev/api/workspace/allocations and verify the endpoint returns a 200 with a response that looks like the photo below.

## UI Photos:
<img width="302" alt="image" src="https://github.com/DesignSafe-CI/portal/assets/43251554/a5969793-e876-47c5-a422-90ba13127d54">


## Notes: ##
- On the form where this hook is used, will need to check allocation shortlist to be used for DS portal against what user has access to.
